functions/src/analytics.ts

```diff
@@ -0,0 +1,48 @@
+import * as functions from 'firebase-functions';
+import * as admin from 'firebase-admin';
+import { IParticipant } from './participant';
+
+type DocumentReference = admin.firestore.DocumentReference;
+
+export const getAnalytics = functions.https.onCall(async (data, context) => {
+  let registeredCount = 0;
+  let attendedCount = 0;
+  let schoolCount = 0;
+  const genderCount = {
+    male: 0,
+    female: 0
+  };
+
+  const db = admin.firestore();
+  const eventDoc = db.collection("events").doc("dragonhacks2019");
```

Comment:
I think it would be wise to make the argument to .doc() an argument to the function so the site can query data from previous years.